runtime/sema/type.go

```diff
@@ -73,6 +63,42 @@ func qualifiedIdentifier(identifier string, containerType Type) string {
 	return sb.String()
 }
 
+func containerTypeNames(typ Type, level int) (typeNames []string, bufSize int) {
+	if typ == nil {
+		return make([]string, level), 0
+	}
+
+	var typeName string
+	var containerType Type
+
+	switch typedContainerType := typ.(type) {
+	case *InterfaceType:
+		typeName = typedContainerType.Identifier
+		containerType = typedContainerType.ContainerType
+	case *CompositeType:
+		typeName = typedContainerType.Identifier
+		containerType = typedContainerType.ContainerType
+	default:
+		switch typ {
+		case PublicAccountType:
+			typeName = string(typedContainerType.ID())
+			containerType = nil
+		case AuthAccountType:
+			typeName = string(typedContainerType.ID())
+			containerType = nil
```

Comment:
Do we have to special case these specific types here? It would be great if we could avoid it

Reply:
(I see now that this was already in the original code, https://github.com/onflow/cadence/pull/948/files#diff-c01765841e64b85306258f5ee4daff2bdc5b9e9e107167f55a6dba4162e8bdb8L50-L55)

Reply:
Yeah, it's simply a move from the original code.

But that's a good catch. Those special cases are baggage coming from pre-refactoring days. Now, those also go under the `CompositeType` path, and no longer needed to be special-cased. Cleaned it up in 1222de9

Reply:
Great, thank you for cleaning this up! 👏